src/parachain/issue.ts

```diff
@@ -89,11 +90,9 @@ export interface IssueAPI extends TransactionAPI {
      * @remarks If `txId` is not set, the `merkleProof` and `rawTx` must both be set.
      *
      * @param issueId The ID returned by the issue request transaction
-     * @param txId (Optional) The ID of the Bitcoin transaction that sends funds to the vault address.
-     * @param merkleProof (Optional) The merkle inclusion proof of the Bitcoin transaction.
-     * @param rawTx (Optional) The raw bytes of the Bitcoin transaction
+     * @param txFetcher TxFetchingDetails object
      */
-    execute(issueId: string, txId?: string, merkleProof?: Bytes, rawTx?: Bytes): Promise<void>;
+    execute(requestId: string, txFetchingDetails: TxFetchingDetails): Promise<void>;
```

Comment:
I'm wondering if we shouldn't just simplify the API at this point by requiring the btcTxId, fetch the proof ourselves and if some really wants to dig deep, we could in a later sprint introduce `executeAdvanced` with extra stuff to set. But at this point we are polluting the API interface for a use case that does not exist (maybe yet) in practice.
Another consideration for me is that this change will have breaking changes on the UI without good reason really.


So my vote is to simplify all of these to only take the `btcTxId` as a required param and that's it. I know that is a change from what we tried to do before, but I think better build things where there is demand for and make our own life easier at this point by reducing complexity.

What do you think? Also @gregdhill do you have an opinion on this?

Reply:
I agree, definitely doesn't make sense to increase complexity for the sake of non-existent downstream users. We've already had this bite us with the `interbtc-api` / `interbtc-js` split.